api/src/main/java/org/openmrs/module/csaudecore/interceptor/entityhandler/util/ObservationUtils.java

```diff
@@ -0,0 +1,156 @@
+package org.openmrs.module.csaudecore.interceptor.entityhandler.util;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.Concept;
+import org.openmrs.Condition;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.csaudecore.util.CSaudeCoreConstants;
+import org.openmrs.module.csaudecore.util.EncounterUtils;
+
+public class ObservationUtils {
+	
+	private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.S");
+	
+	private static final String INACTIVATE_OBS_SQL = "UPDATE obs SET voided = TRUE, voided_by = %d, date_voided = '%s', void_reason = 'voided via POC' WHERE obs_id = %d";
+	
+	private static final String INSERT_NEW_OBS_SQL = "INSERT INTO obs (person_id, concept_id, obs_datetime, location_id, value_coded, value_text, creator, date_created, voided, uuid, encounter_id, previous_version, status) "
+	        + "VALUES (%d, %d, '%s', %d, %s, %s, %d, '%s', 0, '%s', %d, %d, 'AMENDED')";
+	
+	public static void addOrUpdateObsToCondition(Condition condition) {
```

Comment:
Podia-se injetar os serviços do OpenMRS em vez de usar `Context.get...Service` para tornar os testes unitários mais fáceis de implementar.
```suggestion
	private EncounterService encounterService;

	private ConceptService conceptService;

	private AdministrationService administrationService;

	private FormService formService;

	private EncounterUtils encounterUtils;

	@Lazy
	public ObservationUtils(EncounterService encounterService, ConceptService conceptService,
			@Qualifier("adminService") AdministrationService administrationService, FormService formService,
			EncounterUtils encounterUtils) {
		this.encounterService = encounterService;
		this.conceptService = conceptService;
		this.administrationService = administrationService;
		this.formService = formService;
		this.encounterUtils = encounterUtils;
	}

	public void addOrUpdateObsToCondition(Condition condition) {
```

Reply:
@ynurmahomed  actualizado